### PR TITLE
Fixing Codacy issues

### PR DESCRIPTION
**Pull request by hkaiser**

### CODE REVIEW

**Comment by msimberg** [comment]
Thanks @hkaiser for doing this!

**Comment by hkaiser** [comment]
@msimberg Please review, this is all I can stand doing in this direction for now ;-)

File: components/iostreams/tests/regressions/lost_output_2236.cpp [hunk 1]
```diff
@@ -337,8 +329,7 @@ namespace server {
     {
         if (cid > cd->cid)
         {
-            HPX_TEST(
-                "change of owner during recovery not handled" == nullptr);
+            HPX_TEST("change of owner during recovery not handled" == nullptr);
```
Comment (msimberg) [inline components/iostreams/tests/regressions/lost_output_2236.cpp]: Should we maybe change this to `HPX_TEST_MSG(false, "change of owner during recovery not handled")`? (There's another similar one hiding 12 lines above.)

File: examples/quickstart/EasyBMP/EasyBMP.cpp [hunk 2]
```diff
@@ -445,7 +445,6 @@ bool BMP::WriteToFile( const char* FileName )
    cout << "EasyBMP Error: Cannot open file "
         << FileName << " for output." << endl;
   }
-  fclose( fp );
```
Comment (msimberg) [inline examples/quickstart/EasyBMP/EasyBMP.cpp]: Nice!

**Comment by hkaiser** [comment]
I have no idea why the Codacy errors show up _now_. All of them are bogus and I don't know how to disable them. Even more as  the analysis results shown for the commit itself are green: https://app.codacy.com/app/STEllAR-GROUP/hpx/commit?bid=13877137&cid=385608496

**Comment by hkaiser** [comment]
> I have no idea why the Codacy errors show up now. All of them are bogus and I don't know how to disable them. Even more as the analysis results shown for the commit itself are green: https://app.codacy.com/app/STEllAR-GROUP/hpx/commit?bid=13877137&cid=385608496

Ahh, now it came to its senses. Please ignore my comment above.

File: hpx/performance_counters/counters.hpp [hunk 3]
```diff
@@ -42,8 +44,8 @@ namespace hpx { namespace performance_counters
 
     inline std::string& remove_counter_prefix(std::string& name)
     {
-        if (name.find(counter_prefix) == 0)
-            name = name.substr(sizeof(counter_prefix)-1);
+        if (name.compare(0, counter_prefix_len, counter_prefix) != 0)
```
Comment (msimberg) [inline hpx/performance_counters/counters.hpp]:
```suggestion
        if (name.compare(0, counter_prefix_len, counter_prefix) == 0)
```
?

File: hpx/runtime/parcelset/parcelhandler.hpp [hunk 4]
```diff
@@ -194,10 +194,13 @@ namespace hpx { namespace parcelset
...
-                boost::system::error_code const& ec, parcel const & p) -> void {
-                    return invoke_write_handler(ec, p);
-                });
+            auto f =
+                [=](boost::system::error_code const& ec, parcel const& p) -> void
+                {
+                    invoke_write_handler(ec, p);
+                };
+
+            put_parcel(std::move(p), std::move(f));
```
Comment (msimberg) [inline hpx/runtime/parcelset/parcelhandler.hpp]: Is this one again about sequencing of captures and moves within a statement?

File: tools/docker/Dockerfile [hunk 5]
```diff
@@ -6,7 +6,7 @@
 FROM stellargroup/build_env:ubuntu
 
 # docker build needs to be run in /usr/local
-ADD . /usr/local
+COPY . /usr/local
```
Comment (msimberg) [inline tools/docker/Dockerfile]: I guess this change is because `ADD` is overkill? Did codacy actually catch this?

**Comment by msimberg** [comment]
> @msimberg Please review, this is all I can stand doing in this direction for now ;-)

A little is better than nothing, and this is a lot. Thanks a lot!

We should also try the the check for explicit member initializers in `clang-tidy` now and see how much is left. I asked @aurianer to wait with it because there were so many warnings, but now that you've done the hard work...

File: components/iostreams/tests/regressions/lost_output_2236.cpp (hunk 1 above)
Comment (hkaiser) [reply components/iostreams/tests/regressions/lost_output_2236.cpp]: Sure, I can apply this change.

File: tools/docker/Dockerfile (hunk 5 above)
Comment (hkaiser) [reply tools/docker/Dockerfile]: Yes, Codacy caught this. I have no idea what the difference is ;-)

File: hpx/runtime/parcelset/parcelhandler.hpp (hunk 4 above)
Comment (hkaiser) [reply hpx/runtime/parcelset/parcelhandler.hpp]: No, this is about tricking the analysis tools used by Codacy not to complain about non-existent problems ;-)

File: hpx/performance_counters/counters.hpp (hunk 3 above)
Comment (hkaiser) [reply hpx/performance_counters/counters.hpp]: Excellent catch!

**Comment by hkaiser** [comment]
@msimberg I think this is ready to go. I still don't understand why this shows the wrong Codacy results, though. The correct results are here: https://app.codacy.com/app/STEllAR-GROUP/hpx/pullRequest?prid=4008999.

**Comment by hkaiser** [comment]
> @msimberg I think this is ready to go. I still don't understand why this shows the wrong Codacy results, though. The correct results are here: https://app.codacy.com/app/STEllAR-GROUP/hpx/pullRequest?prid=4008999.

I think I found the problem. We did have two independent webhooks registered for Codacy. I removed one of those, that should fix the problem in the future.

**Pull request merged**